integration/tests/params.go

```diff
@@ -217,3 +226,44 @@ func NewParams(clusterNamePrefix string) *Params {
 	params.GenerateCommands()
 	return &params
 }
+
+// attemptSettingUserID will attempt to fetch the first 2 characters of the userID running the
+// integration tests to be used as a prefix for cluster names. It fails silently if an error occurs
+func (p *Params) attemptSettingUserID() {
+	clusterProvider, err := eks.New(&api.ProviderConfig{Region: p.Region}, &api.ClusterConfig{
+		TypeMeta: api.ClusterConfigTypeMeta(),
+		Metadata: &api.ClusterMeta{
+			Version: p.Version,
+			Name:    p.ClusterName,
+			Region:  p.Region,
+		},
+	})
+
+	msg := "Warning: failed to get UserID, proceeding without configuring cluster name prefix. Error: %w"
+	if err != nil {
+		fmt.Fprintf(GinkgoWriter, msg, err)
+		return
+	}
+
+	out, err := clusterProvider.Provider.STS().GetCallerIdentity(&sts.GetCallerIdentityInput{})
+	if err != nil {
+		fmt.Fprintf(GinkgoWriter, msg, err)
+		return
+	}
+
+	if out.UserId == nil {
+		fmt.Fprintf(GinkgoWriter, msg, fmt.Errorf("UserId value not set in AWS response"))
+		return
+	}
+
+	//example value: ABCD1234:my-name@email-provider
+	parts := strings.Split(*out.UserId, ":")
+	if len(parts) == 2 {
+		parts = strings.Split(parts[1], "@")
+		if len(parts) == 2 {
+			p.userID = parts[0][:4]
```

Comment:
For the future, names that are below 4 characters are definitely there. Especially names like Ale, Joy, Hub, Xi, Lo, etc, or even usernames that aren't long enough such as cpu. This will then blow up with an index out of bounds exception. That's why I thought 2 characters is enough for this.

Reply:
true, if/when this happens we can reassess